### DISCUSSION TITLE
Does this not work with the updated databases?

### DISCUSSION THREAD

**Discussion by Ravie13** [mathix420/notion-charts#25] [Q&A] [answered]
Hi there - this looks like just the code I need!
I followed the directions for a private notion page and ran it but I'm getting errors in the console. I'm wondering if it is because of the recent Notion Database updates?

```
Failed to load resource: the server responded with a status of 500 ()
(index):203 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'map')
    at (index):203:34
```

At that index in the code this is the function:
```

 json.columns.map(function (column) {
                        const opt = document.createElement('option');
                        opt.innerText = column.name;
                        opt.value = column.id;
                        return opt;
                    }).forEach(function (child) {
                        fields.appendChild(child);
                        firstColumnSav = firstColumn.cloneNode(true);
                        firstColumnSav.querySelector('input').value = '';
                    });
```

**Comment by mathix420** [comment, marked as answer]
Hi Ravie!

Thanks for bringing this to my attention.
It seems you're right, `notion-charts` can't handle the new database paradigm for now..

I will try to fix this soon, but I have quite a lot of work these days.
So a contribution will be more than welcome :yum: 

In any case thanks again and have a nice day/evening!
